src/Widgets/GoogleMapsWidget/GoogleMapsWidgetComponent.js

```diff
@@ -7,129 +7,128 @@ import "./GoogleMapsWidget.scss";
 
 const maxWidth = 640;
 
-class GoogleMapsWidgetComponent extends React.Component {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      elementHeight: 0,
-      elementWidth: 0,
-      height: null,
-      width: null,
-    };
-
-    this.outerDivRef = React.createRef();
-
-    this.handleResize = this.handleResize.bind(this);
-  }
-
-  componentDidMount() {
-    this.handleResize();
-    window.addEventListener("resize", this.handleResize);
-  }
-
-  componentWillUnmount() {
-    window.removeEventListener("resize", this.handleResize);
-  }
+function GoogleMapsWidgetComponent({ widget }) {
+  const address = widget.get("address") || "Brandenburg Gate, Berlin, Germany";
+  const zoom = widget.get("zoom") || "15";
+  const apiKey = googleMapsApiKey();
+  const mapType = widget.get("mapType") || "static";
 
-  handleResize() {
-    const elementWidth = this.outerDivRef.current.offsetWidth;
-    const elementHeight = this.outerDivRef.current.offsetHeight;
+  const Map = mapType === "static" ? StaticGoogleMap : InteractiveGoogleMap;
 
-    if (
-      this.state.elementWidth !== elementWidth ||
-      this.state.elementHeight !== elementHeight
-    ) {
-      let width = elementWidth;
-      let height = elementHeight;
+  return (
+    <Map address={address} zoom={zoom} apiKey={apiKey}>
+      <Widgets widget={widget} mapType={mapType} />
+    </Map>
+  );
+}
 
-      if (width > maxWidth) {
-        width = maxWidth;
+function InteractiveGoogleMap({ address, apiKey, zoom, children }) {
+  const url = `https://www.google.com/maps/embed/v1/place?q=${address}&key=${apiKey}&zoom=${zoom}`;
+  return (
+    <div className="google-maps-widget">
+      <iframe
+        title="Interactive Map"
+        frameBorder="0"
+        style={{ border: 0 }}
+        src={url}
+        loading="lazy"
+      />
+      {children}
+    </div>
+  );
+}
 
-        const factor = elementHeight / elementWidth;
-        height = Math.round(maxWidth * factor);
+function StaticGoogleMap({ address, apiKey, zoom, children }) {
+  const [mapSize, setMapSize] = React.useState({
+    elementHeight: 0,
+    elementWidth: 0,
+    height: null,
+    width: null,
+  });
+
+  const ref = React.useRef(null);
```

Comment:
Using `useRef` with `useEffect` is discouraged, because it may cause issues when passing the ref to child components.  ([Hooks FAQ](https://reactjs.org/docs/hooks-faq.html#how-can-i-measure-a-dom-node)).

* [ ] FYI I'd recommend to stick with the recommended `useCallback` approach to allow for code changes without running into subtle issues.